File: algorithms/algorithms-cplusplus/leetcode/Question_0453_Minimum_Moves_to_Equal_Array_Elements.cc
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
 public:
  int minMoves(vector<int>& nums) {
    if (nums.size() <= 1) {
      return 0;
    }
    long sum = 0, n = nums[0];
    for (long num : nums) {
      n = min(n, num);
      sum += num;
    }
    return (int)(sum - n * nums.size());
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0453_Minimum_Moves_to_Equal_Array_Elements.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0453_Minimum_Moves_to_Equal_Array_Elements {
    public int minMoves(int[] nums) {
        if (nums == null || nums.length <= 1) {
            return 0;
        }
        long sum = 0, n = nums[0];
        for (long num : nums) {
            sum += num;
            n = Math.min(num, n);
        }
        return (int)(sum - n * nums.length);
    }
}
